pandas/core/accessors.py

```diff
@@ -0,0 +1,345 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+
+An example/recipe for creating a custom accessor.
```

Comment:
> example seems a little clunky compared to what xarray does

The examples are for pretty different use cases.  The xarray example defines a `center` attribute in terms of multiple existing [column]s of a [DataFrame].  That's why it doesn't need any vectorization boilerplate.

The example I used vectorizes properties/methods of the elements in a Series.  `StringMethods` or `CategoricalAccessor` work roughly this way.  (`CombinedDatetimelikeProperties` doesn't need to apply the property/method point-wise because they already exist in the underlying Series/Index.)

That said, if this isn't already obvious, that is a shortcoming of the documentation.

There is some avoidable clunkiness in that we are requiring every accessible attribute to be explicitly listed.